libnmstate/nm/connection.py

```diff
@@ -294,6 +300,28 @@ def waitfor_active_connection_async(self, ac):
                 "state-changed", self._waitfor_device_state_change_callback, ac
             )
         )
+        # As we also depend on IP4_READY flag which does not trigger
+        # 'state-change' signal, hence do a repeated recheck without timeout.
```

Comment:
Isn't this an NM bug? If the flags changed, I would expect the ac get a signal about it.